components/error_code/src/lib.rs

```diff
@@ -37,7 +43,20 @@ pub struct ErrorCode {
 
 impl Display for ErrorCode {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "[{}] {}", self.code, self.description)
+        write!(f, "{}", self.code)
+    }
+}
+
+impl Serialize for ErrorCode {
```

Comment:
Why impl by hand instead of derive `Serialize`?

Reply:
To change the field name.